python-checks/src/main/java/org/sonar/python/checks/tests/AssertOnDissimilarTypesCheck.java

```diff
@@ -81,18 +84,20 @@ private static void checkArguments(SubscriptionContext ctx, CallExpression callE
     }
 
     List<Argument> arguments = args.arguments();
-    if (!(arguments.size() >= 2 && arguments.get(0).is(Tree.Kind.REGULAR_ARGUMENT) && arguments.get(1).is(Tree.Kind.REGULAR_ARGUMENT))) {
+    RegularArgument firstArg = TreeUtils.nthArgumentOrKeyword(0, FIRST_ARG_KEYWORD, arguments);
+    RegularArgument secondArg = TreeUtils.nthArgumentOrKeyword(1, SECOND_ARG_KEYWORD, arguments);
+    if (firstArg == null || secondArg == null) {
       return;
     }
 
-    Expression left = ((RegularArgument) arguments.get(0)).expression();
-    Expression right = ((RegularArgument) arguments.get(1)).expression();
-    if (!checkArgumentsCantBeIdentical(left, right)) {
+    Expression left = firstArg.expression();
+    Expression right = secondArg.expression();
+    if (canArgumentsBeIdentical(left, right)) {
       return;
     } else if (isAnAssertIdentity) {
       shouldRaise = true;
     }
-    if (isAnAssertEquality && checkArgumentsCantBeEqual(left, right)) {
+    if (isAnAssertEquality && !canArgumentsBeEqual(left, right)) {
       shouldRaise = true;
     }
```

Comment:
I just realize we can further simplify this by using a single:
```
if (isAnAssertIdentity || !canArgumentsBeEqual(left, right)) {
    shouldRaise = true;
}
```
Since at this point, if we're in an identity assertion, we know we must raise anyway, and if we're not, that means we are in an assert equality.